sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1863,6 +1863,17 @@ object functions extends LegacyFunctions {
    */
   def crc32(e: Column): Column = withExpr { Crc32(e.expr) }
 
+  /**
+   * Calculates the hash code of given columns, and returns the result as a int column.
+   *
+   * @group misc_funcs
+   * @since 2.0
+   */
+  @scala.annotation.varargs
+  def hash(col: Column, cols: Column*): Column = withExpr {
```

Comment:
this should just be  a single column vararg, rather than one followed by vararg?